src/nanodbc.cpp

```diff
@@ -2348,6 +2348,38 @@ class result::result_impl
         return col.sqltype_;
     }
 
+    string_type column_datatype_name(short column) const
+    {
+        if (column >= bound_columns_size_)
+            throw index_range_error();
+
+        NANODBC_SQLCHAR type_name[256] = {0};
+        SQLSMALLINT len = 0; // total number of bytes
+        RETCODE rc;
+        NANODBC_CALL_RC(
+            SQLColAttribute,
+            rc,
+            stmt_.native_statement_handle(),
+            column + 1,
+            SQL_DESC_TYPE_NAME,
+            type_name,
+            sizeof(type_name) / sizeof(NANODBC_SQLCHAR),
+            &len,
+            nullptr);
+        if (!success(rc))
+            NANODBC_THROW_DATABASE_ERROR(stmt_.native_statement_handle(), SQL_HANDLE_STMT);
+
+        NANODBC_ASSERT(len % sizeof(NANODBC_SQLCHAR) == 0);
+        len = len / sizeof(NANODBC_SQLCHAR);
+        return string_type(type_name, type_name + len);
+    }
+
+    string_type column_datatype_name(const string_type& column_name) const
+    {
+        const short column = this->column(column_name);
+        return column_datatype_name(column);
```

Comment:
I suppose these two lines could be made into one line but I don't think it matters either way.